src/components/organisms/AssetActions/Compute/index.tsx

```diff
@@ -0,0 +1,300 @@
+        gueryResults.results,
+        config.metadataCacheUri,
+        []
+      )
+    }
+    return algorithmSelectionList
+  }
+
+  useEffect(() => {
+    if (!ddo) return
+    getAlgorithmList().then((algorithms) => {
+      setAlgorithmList(algorithms)
+    })
+  }, [])
+
+  useEffect(() => {
+    if (!ocean || !accountId) return
+    checkPreviousOrders(ddo, 'compute')
+  }, [ocean, ddo, accountId])
+
+  useEffect(() => {
+    if (!ocean || !accountId || !selectedAlgorithmAsset) return
+    checkPreviousOrders(selectedAlgorithmAsset, 'access')
+    checkAssetDTBalance(selectedAlgorithmAsset)
+  }, [selectedAlgorithmAsset])
+
+  const startJob = async (algorithmId: string) => {
+    try {
+      if (!ocean) return
+
+      setIsJobStarting(true)
+      setIsPublished(false)
+      setError('')
+
+      const computeService = ddo.findServiceByType('compute')
+      const serviceAlgo = selectedAlgorithmAsset.findServiceByType('access')
+
+      !hasPreviousDatasetOrder &&
+        !hasDatatoken &&
+        (await buyDT('1', (ddo as DDO).price, ddo))
+
+      !hasPreviousAlgorithmOrder &&
+        !hasAlgoAssetDatatoken &&
+        (await buyDT(
+          '1',
+          (selectedAlgorithmAsset as DDO).price,
+          selectedAlgorithmAsset
+        ))
+
+      const allowed = await ocean.compute.isOrderable(
+        ddo.id,
+        computeService.index
+      )
+
+      console.log('is dataset orderable ?', allowed)
+      const assetOrderId = hasPreviousDatasetOrder
+        ? previousDatasetOrderId
+        : await ocean.compute.orderAsset(
+            accountId,
+            ddo.id,
+            computeService.index
+          )
+
+      const algorithmAssetOrderId = hasPreviousAlgorithmOrder
+        ? previousAlgorithmOrderId
+        : await ocean.compute.orderAlgorithm(
+            algorithmId,
+            serviceAlgo.type,
+            accountId,
+            serviceAlgo.index
+          )
+
+      const output = {}
+      const respone = await ocean.compute.start(
+        ddo.id,
+        assetOrderId,
+        ddo.dataToken,
+        account,
+        algorithmId,
+        undefined,
+        output,
+        `${computeService.index}`,
+        computeService.type,
+        algorithmAssetOrderId,
+        selectedAlgorithmAsset.dataToken
+      )
+      console.log('respone', respone)
+
+      setHasPreviousDatasetOrder(true)
+      setIsPublished(true)
+    } catch (error) {
+      console.log('error found')
+      setError('Failed to start job!')
+      Logger.error(error.message)
+    } finally {
+      setIsJobStarting(false)
+    }
+  }
+
+  async function handleSubmit(
+    values: any,
+    resetForm: () => void
+  ): Promise<void> {
+    await startJob(values.algorithm)
+  }
+
+  return (
+    <>
+      <div className={styles.info}>
+        <File file={file} small />
+        <Price price={(ddo as DDO).price} conversion />
+      </div>
+
+      {type === 'algorithm' ? (
+        <Input
```

Comment:
I'm in favor of removing this functionality from the current scope for timing reasons. We can replace this with some info message directing users to data sets to run compute jobs or something